src/BrefKernel.php

```diff
@@ -24,6 +29,29 @@ public function getCacheDir()
         return parent::getCacheDir();
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function getBuildDir(): string
+    {
+        $buildDir = $this->getProjectDir().'/var/build/'.$this->environment;
+
+        if ($this->isLambda() && !is_dir($buildDir)) {
+            if (!$this->allowEmptyBuildDir()) {
+                throw new \Exception(
+                    'You must warm and deploy the build directory as part of your Lambda package
+                    as this directory is readonly on Lambda.
+                    You can alternatively define the function allowEmptyBuildDir() in your Kernel class
+                    and return "true" if you still want to deploy without a warm build directory.
+                ');
```

Comment:
If we return `true`, the app will break right? (because the build directory is read-only, it cannot be warmed)

Reply:
If we return true, the exception is not thrown so it continues to line 49 `return $this->getCacheDir();` which will eventually resolve to `/tmp/cache` so it will not error but warm the cache and build at runtime on each new lambda.

Reply:
This allows for people that don't care about performance or just want to quickly test without bothering on deploying a properly warmed cache to do it. 
But as it's not the recommend approach I prefer it to be false by default and throw an explicit exception.

This would also need to be documented and is just a proposal, not a definitive opinion of mine.

Reply:
🤦  you're right, no exception/error here.

Personally, I'm a big fan of being able to deploy without any step (i.e. without cache generation). It's important for the DX, especially when testing out.

But because of that preference, I've slowed the effort of other contributors in the past. Now I think it's more important to get it working well for Symfony 5.2 in good condition, and maybe later iterate on that and allow deploying without cache.

Reply:
No worries, I think thats good discussions to have but I agree that we can iterate to .